bpo-36160: Fix test_site so that it can run independently of other tests

native-api opened the pull request:
<!-- issue-number: [bpo-36160](https://bugs.python.org/issue36160) -->
https://bugs.python.org/issue36160
<!-- /issue-number -->

native-api committed:
Fix AttributeError on sysconfig._CONFIG_VARS.clear() if test_site is run separately

native-api committed:
Merge branch 'master' into test_site_configvars

native-api committed:
(patchcheck fix writespace)

remilapeyre reviewed (changes requested):
Hi @native-api, thanks for the patch but it looks like it has been fixed in another PR:

```
./python -m test.test_site
................s.s.........ss
----------------------------------------------------------------------
Ran 30 tests in 0.581s

OK (skipped=4)
```

I think we can close this one.

iritkatriel commented on Lib/test/test_site.py:
```diff
@@ -49,6 +49,8 @@ def setUpModule():
                                     % (site.USER_SITE, exc))
 
+    # sysconfig._CONFIG_VARS is None until the first call to this function
+    sysconfig.get_config_vars()
```
I think it would be better to guard the `sysconfig._CONFIG_VARS.clear()` call in tearDown rather than add an unnecessary initialization.

native-api replied:
Sounds good. Accessing `sysconfig._CONFIG_VARS` directly is an unsupported operation -- so it's the job of whoever does it to make sure that it works in all cases.

native-api committed:
Guard the unsupported operation instead

Accessing _CONFIG_VARS directly is unsupported so it's the job of whoever does
that to ensure that it works in all cases

native-api committed:
formatting

iritkatriel committed:
no need for news in this case

iritkatriel committed:
Comment line too long

iritkatriel committed:
fix comment
